File: json/json-annotation-processor/src/test/java/io/koraframework/json/annotation/processor/dto/DtoWithNullableFields.java
package io.koraframework.json.annotation.processor.dto;

import org.jspecify.annotations.Nullable;
import io.koraframework.json.common.annotation.JsonField;
import io.koraframework.json.common.annotation.JsonReader;

@JsonReader
public record DtoWithNullableFields(@JsonField("field_1") String field1, int field4, @Nullable String field2, @Nullable String field3) {}

File: json/json-symbol-processor/src/test/kotlin/io/koraframework/json/ksp/dto/DtoWithNullableFields.kt
package io.koraframework.json.ksp.dto

import io.koraframework.json.common.annotation.JsonField
import io.koraframework.json.common.annotation.JsonReader

@JsonReader
data class DtoWithNullableFields(
    @JsonField("field_1") val field1: String,
    val field4: Int,
    val field2: String?,
    val field3: String?
)
